### PR TITLE
add dev.ensoft.ecode

### PR DESCRIPTION
**Pull request by proletarius101**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:** https://github.com/SpartanJ/eepp/pull/13#issuecomment-1422623055
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

File: dev.ensoft.ecode.yml [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+modules:
+  - shared-modules/SDL2/SDL2-with-libdecor.json
+
+  - name: premake5
+    buildsystem: simple
+    build-commands:
+      - make -f Bootstrap.mak linux
+      - install -Dm755 -t "${FLATPAK_DEST}/bin" bin/release/premake5
+    cleanup:
+      - /bin
```
Comment (hfiguiere) [inline dev.ensoft.ecode.yml:26]:
```suggestion
      - '*'
```

File: dev.ensoft.ecode.yml [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+rename-icon: ecode
+command: ecode
+finish-args:
+  - --device=dri
+  - --share=ipc
+  - --socket=wayland
+  - --socket=fallback-x11
+
+build-options:
+  prepend-path: /app/bin
```
Comment (hfiguiere) [inline dev.ensoft.ecode.yml:15]:
```suggestion
```

You don't need this.

File: dev.ensoft.ecode.yml [hunk 3]
```diff
@@ -0,0 +1,78 @@
+app-id: dev.ensoft.ecode
```
Comment (hfiguiere) [inline dev.ensoft.ecode.yml:1]: `ensoft.dev` is mentionned nowhere and the website is broken. Is this really upstreams?

Comment (hfiguiere) [reply dev.ensoft.ecode.yml:1]: so why isn't the homepage on the appstream file set to it?

Comment (SpartanJ) [reply dev.ensoft.ecode.yml:1]: Hi! Author of ecode here. There's no landing page for the project at the moment. But I'm using that domain to host some documentation and the web version.
You can see the domain is mentioned in: https://github.com/SpartanJ/ecode#live-demo, the demo is located at: https://cdn.ensoft.dev/eepp-demos/demo-fs.html?run=ecode.js

Comment (SpartanJ) [reply dev.ensoft.ecode.yml:1]: Give me some minutes and I'll register `ecode.ensoft.dev` and redirect to `https://github.com/SpartanJ/ecode`. I would love to have `ecode.dev` but it's registered, I'll take a look for a domain for the project.

Comment (hfiguiere) [reply dev.ensoft.ecode.yml:1]: it's not blocker. Just that there are so many people you decided using random domain that I have to double check.

Comment (SpartanJ) [reply dev.ensoft.ecode.yml:1]: `ecode.ensoft.dev` is already working

**Comment by SpartanJ** [comment]
It's not detecting correctly aarch64 and the build will fail, please give me some time to push a fix.

**Comment by SpartanJ** [comment]
@proletarius101 v0.4.5 with aarch64 support has been released. You should be able to build that version for flathub.

**Comment by SpartanJ** [comment]
I think I did not receive the invitation.
A new version of ecode has been released, I created a pull request [here](https://github.com/flathub/dev.ensoft.ecode/pull/1). It's OK?

**Pull request closed without merging**